ground/src/main/java/com/google/android/ground/ui/common/AbstractMapContainerFragment.kt

```diff
@@ -18,62 +18,60 @@ package com.google.android.ground.ui.common
 import android.os.Bundle
 import android.view.View
 import android.widget.Toast
+import androidx.lifecycle.Lifecycle
 import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
 import com.google.android.ground.R
-import com.google.android.ground.rx.RxAutoDispose
+import com.google.android.ground.coroutines.DefaultDispatcher
 import com.google.android.ground.system.GeocodingManager
 import com.google.android.ground.system.PermissionDeniedException
 import com.google.android.ground.system.SettingsChangeRequestCanceled
 import com.google.android.ground.ui.home.mapcontainer.MapTypeDialogFragmentDirections
 import com.google.android.ground.ui.map.CameraPosition
-import com.google.android.ground.ui.map.Map
+import com.google.android.ground.ui.map.MapView
 import javax.inject.Inject
 import kotlin.math.max
+import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.launch
 import timber.log.Timber
 
-/** Injects a [Map] in the container with id "map" and provides shared map functionality. */
+/** Injects a [MapView] in the container with id "map" and provides shared map functionality. */
 abstract class AbstractMapContainerFragment : AbstractFragment() {
 
-  @Inject lateinit var map: Map
+  @Inject lateinit var map: MapView
   @Inject lateinit var navigator: Navigator
   @Inject lateinit var geocodingManager: GeocodingManager
+  @Inject @DefaultDispatcher lateinit var defaultDispatcher: CoroutineDispatcher
 
   override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
     super.onViewCreated(view, savedInstanceState)
     map.attachToFragment(this, R.id.map) { onMapAttached(it) }
   }
+  private fun launchWhenStarted(fn: suspend () -> Unit) {
+    lifecycleScope.launch { repeatOnLifecycle(Lifecycle.State.STARTED) { fn.invoke() } }
+  }
 
-  private fun onMapAttached(map: Map) {
+  private fun onMapAttached(map: MapView) {
     val viewModel = getMapViewModel()
 
     // Removes all markers, overlays, polylines and polygons from the map.
     map.clear()
 
-    map.cameraMovedEvents
-      .onBackpressureLatest()
-      .`as`(RxAutoDispose.disposeOnDestroy(this))
-      .subscribe { onMapCameraMoved(it) }
-    map.startDragEvents
-      .onBackpressureLatest()
-      .`as`(RxAutoDispose.disposeOnDestroy(this))
-      .subscribe { viewModel.onMapDragged() }
-
-    lifecycleScope.launch {
-      getMapViewModel().locationLock.collect { onLocationLockStateChange(it, map) }
-    }
-    lifecycleScope.launch {
+    launchWhenStarted { map.cameraMovedEvents.collect { viewModel.onMapCameraMoved(it) } }
+    launchWhenStarted { map.startDragEvents.collect { viewModel.onMapDragged() } }
+    launchWhenStarted { viewModel.locationLock.collect { onLocationLockStateChange(it, map) } }
+    launchWhenStarted {
```

Comment:
What would happen when the map is attached again (e.g. when pausing and then resuming the fragment)? Would the previous coroutine be cancelled?

Reply:
Yes, `repeatOnLifecycle` does this for us. I just learned we can now move these initializations to the constructor and replace our `launchWhenStarted` helper with `whenStarted`, but out of scope of this PR.